src/ray/pubsub/subscriber.h

```diff
@@ -188,37 +202,73 @@ class SubscriberChannel {
+  /// information is received by the subscriber.
+  /// \param subscription_failure_callback A callback that is invoked whenever the
+  /// connection to publisher is broken (e.g. the publisher fails).
+  virtual bool Subscribe(std::unique_ptr<rpc::SubMessage> sub_message,
+                         rpc::ChannelType channel_type,
                          const rpc::Address &publisher_address, const std::string &key_id,
-                         SubscriptionCallback subscription_callback,
+                         SubscribeDoneCallback subscribe_done_callback,
+                         SubscriptionItemCallback subscription_callback,
                          SubscriptionFailureCallback subscription_failure_callback) = 0;
 
-  /// Unsubscribe the object.
+  /// Subscribe to all entities in channel channel_type.
+  ///
+  /// \param sub_message The subscription message.
+  /// \param channel_type The channel to subscribe to.
+  /// \param publisher_address Address of the publisher to subscribe the object.
+  /// \param subscription_callback A callback that is invoked whenever an entity
+  /// information is received by the subscriber.
+  /// \param subscription_failure_callback A callback that is invoked whenever the
+  /// connection to publisher is broken (e.g. the publisher fails).
+  virtual bool SubscribeAll(
+      std::unique_ptr<rpc::SubMessage> sub_message, rpc::ChannelType channel_type,
+      const rpc::Address &publisher_address,
+      SubscribeDoneCallback subscribe_done_callback,
+      SubscriptionItemCallback subscription_callback,
+      SubscriptionFailureCallback subscription_failure_callback) = 0;
+
+  /// Unsubscribe the entity if the entity has been subscribed with Subscribe().
   /// NOTE: Calling this method inside subscription_failure_callback is not allowed.
   ///
-  /// \param channel_type The channel to unsubscribe to.
-  /// \param publisher_address The publisher address that it will unsubscribe to.
+  /// \param channel_type The channel to unsubscribe from.
+  /// \param publisher_address The publisher address that it will unsubscribe from.
   /// \param key_id The entity id to unsubscribe.
+  /// \return Returns whether the entity key_id has been subscribed before.
   virtual bool Unsubscribe(const rpc::ChannelType channel_type,
                            const rpc::Address &publisher_address,
                            const std::string &key_id) = 0;
 
-  /// Checks if the entity key_id is being subscribed to.
+  /// Unsubscribe from the channel_type. Must be paired with SubscribeAll().
+  /// NOTE: Calling this method inside subscription_failure_callback is not allowed.
+  ///
+  /// \param channel_type The channel to unsubscribe from.
+  /// \param publisher_address The publisher address that it will unsubscribe from.
+  /// \return Returns whether the entity key_id has been subscribed before.
+  virtual bool Unsubscribe(const rpc::ChannelType channel_type,
```

Comment:
Consider UndubscribeAll if it should be paired with SubscribeAll?

Reply:
I felt `UnsubscribeAll()` seems it can unsubscribe from previous `Subscribe()` calls, which it cannot. Renamed the functions to `SubscribeChannel()` / `UnsubscribeChannel()`.